From af0d4f5ba5da81e2f900a1487eb2700d14e92dc6 Mon Sep 17 00:00:00 2001
From: "Joseph Zikusooka (ZIK)" <zik@jambula.net>
Date: Sat, 26 Apr 2025 10:22:04 +0300
Subject: [PATCH] Kea: new package - kea version 2.6.2

Add Kea, an open-source Dynamic Host Configuration Protocol (DHCP)
server developed by the Internet Systems Consortium (ISC).
It's designed to provide dynamic IP address allocation and
configuration for devices on a network. Kea is known for its
modular architecture, allowing for easy extension and integration
with other services.

Supersedes: a015291dc94b5497ac1fd60a2e4cf84fe3c82cb5
---
 package/kea/Config.in | 17 +++++++++++++++++
 package/kea/kea.hash  |  3 +++
 package/kea/kea.mk    | 22 ++++++++++++++++++++++
 3 files changed, 42 insertions(+)
 create mode 100644 package/kea/Config.in
 create mode 100644 package/kea/kea.hash
 create mode 100644 package/kea/kea.mk

diff --git a/package/kea/Config.in b/package/kea/Config.in
new file mode 100644
index 0000000000..1b3bbe2e50
--- /dev/null
+++ b/package/kea/Config.in
@@ -0,0 +1,17 @@
+config BR2_PACKAGE_KEA
+
+	bool "kea"
+	depends on BR2_TOOLCHAIN_HAS_THREADS
+	depends on BR2_USE_MMU
+	select BR2_PACKAGE_BOOST
+	select BR2_PACKAGE_OPENSSL
+	select BR2_PACKAGE_LOG4CPLUS
+
+	help
+	  DHCP implementation from Internet Systems Consortium, Inc. that features fully
+	  functional DHCPv4, DHCPv6 and Dynamic DNS servers.
+	  Both DHCP servers fully support server discovery, address assignment, renewal,
+	  rebinding and release. The DHCPv6 server supports prefix delegation. Both
+	  servers support DNS Update mechanism, using stand-alone DDNS daemon
+
+          https://www.isc.org/kea
diff --git a/package/kea/kea.hash b/package/kea/kea.hash
new file mode 100644
index 0000000000..37cb4e19f4
--- /dev/null
+++ b/package/kea/kea.hash
@@ -0,0 +1,3 @@
+# Locally computed - sha256
+sha256  8a50b63103734b59c3b8619ccd6766d2dfee3f02e3a5f9f3abc1cd55f70fa424  kea-2.6.2.tar.gz
+sha256  2768ea1cbc79ac048f0c34c5cb50cf7e7d4c13a83337ebfa147a700b876431d5  COPYING
diff --git a/package/kea/kea.mk b/package/kea/kea.mk
new file mode 100644
index 0000000000..d589d2c1c2
--- /dev/null
+++ b/package/kea/kea.mk
@@ -0,0 +1,22 @@
+################################################################################
+#
+# kea
+#
+################################################################################
+
+KEA_VERSION = 2.6.2
+KEA_SOURCE = kea-$(KEA_VERSION).tar.gz
+KEA_SITE = https://dl.cloudsmith.io/public/isc/kea-2-6/raw/versions/$(KEA_VERSION)
+KEA_LICENSE = MPL 2.0
+KEA_LICENSE_FILES = COPYING
+
+KEA_DEPENDENCIES = host-pkgconf openssl boost log4cplus
+
+KEA_CONF_OPTS = "--with-openssl=$(STAGING_DIR)/usr" 
+KEA_CONF_OPTS = "--with-log4cplus=$(STAGING_DIR)/usr"
+#KEA_CONF_OPTS = "--with-boost-include=$(STAGING_DIR)/usr"
+
+KEA_INSTALL_STAGING = YES
+KEA_INSTALL_TARGET = YES
+
+$(eval $(autotools-package))
-- 
2.49.0

